public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx

```diff
@@ -146,6 +146,18 @@ export const ConnectionSVG = ({
             yStart = v.sourceOriginal.y;
             xEnd = v.targetOriginal.x;
             yEnd = v.targetOriginal.y;
+          } else if (source.options.connections) {
+            const currentConnections = [...source.options.connections];
+            // Check for original state
+            if (!currentConnections[index].sourceOriginal || !currentConnections[index].targetOriginal) {
+              currentConnections[index] = {
+                ...currentConnections[index],
+                sourceOriginal: { x: x1, y: y1 },
+                targetOriginal: { x: x2, y: y2 },
+              };
+              // Update model with populated originals
+              source.onChange({ ...source.options, connections: currentConnections });
```

Comment:
I am wondering if this type of one time logic would make better sense as migration code vs existing inside the render logic of connections indefinitely 🤔

Reply:
Yeah, great suggestion! Let me look into moving it.

Reply:
I'm still unable to figure out how to access scene.connections.state from inside of migration. Open to suggestions here.

Reply:
Hm I looked more into this and yeah it seems like sticking this logic in a migration is not feasible given the data is based on runtime html dimensions and this logic run before the canvas panel is rendered. For future reference the properties of connections are accessible / changeable in migrations, but not in cases where the values are driven by runtime DOM placement.

```
const root = panel.options?.root;
if (root?.elements) {
  for (const element of root.elements) {
    if (element.connections) {
        for (const connection of element.connections) {
                // Logic here modifying connection properties
        }
      }
    }
}
```